[feat + fix]: course curriculum section + various larger-screen-sizes cleanup

alyssia-x opened the pull request:
# Description
Adding new Course Curriculum Section component with expandable/collapsible unit cards displayed in an accordion-style, fetching from Airtable db.  Responsive for desktop and mobile. 

Standardized upper padding pattern across sections for larger screen sizes (course details, faq, graduates section). Reduced upper padding for course benefits section to-spec now that the course curriculum section has been added.

**Out of scope for this PR:**
- Displaying progress-related features (blocked, will create new PR)

**Note:** I still want to test the custom scrollbar in mobile staging.

## Issue
Implements part of #1334 

## Developer checklist

- [X] Front-end code follows [Component Accessibility Checklist (for Testing)](https://github.com/bluedotimpact/bluedot/blob/master/DEVELOPMENT_HANDBOOK.md#component-accessibility-checklist)
- [X] Considered having snapshot tests and/or happy path functional tests
- [X] Considered adding Storybook stories

## Screenshots
### Desktop
<img width="1512" height="723" alt="course-curriculum" src="https://github.com/user-attachments/assets/f882867b-1733-4b47-8dbf-ac7eb785ee3f" />

![curriculum-overview-desktop](https://github.com/user-attachments/assets/0e47d7be-9d94-4207-b0e4-fab4d2d77d51)

### Mobile
![course-curriculum-mobile](https://github.com/user-attachments/assets/2ea8cf80-4d50-45b8-a7b3-2f3fc7db8ca7)

somepersonnamedjosh commented:
@claude review pls

marn-in-prod commented on apps/website/src/globals.css:
```diff
@@ -42,6 +42,36 @@
   font-family: var(--font-inter-display), var(--font-inter), ui-serif, Georgia, Cambria, 'Times New Roman', Times, serif;
 }
 
+/* Hide default scrollbar while keeping scroll functionality */
+.scrollbar-none {
+  scrollbar-width: none; /* Firefox */
+  -ms-overflow-style: none; /* IE and Edge */
+}
+.scrollbar-none::-webkit-scrollbar {
+  display: none; /* Chrome, Safari and Opera */
+}
+
+/* Custom curriculum scrollbar */
+.scrollbar-curriculum {
+  /* Firefox */
+  scrollbar-width: thin;
+  scrollbar-color: rgba(0, 0, 0, 0.2) transparent;
+}
+.scrollbar-curriculum::-webkit-scrollbar {
+  width: 4px;
+}
+.scrollbar-curriculum::-webkit-scrollbar-track {
+  background: transparent;
+}
+.scrollbar-curriculum::-webkit-scrollbar-thumb {
+  background: rgba(0, 0, 0, 0.2);
+  border-radius: 4px;
+  min-height: 30px;
+}
+.scrollbar-curriculum::-webkit-scrollbar-thumb:hover {
+  background: rgba(0, 0, 0, 0.2);
+}
+
```
Does this make more sense to be in one of the agi-strategy component files instead of inside of global.css? Some of this is very specific to Agi-strategy at the end of the day

alyssia-x replied:
Update: ended up in-lining it!

alyssia-x commented:
> I think the coderabbit suggestion to make sure you don't import directly from the backend, would also be good.
> 
> i.e. addressing the comment
> 
> ```
> -import { GetCourseResponse } from '../../../pages/api/courses/[courseSlug]';
> +import type { GetCourseResponse } from '../../../pages/api/courses/[courseSlug]';
> ```

This was already updated

alyssia-x commented:
**Note** - the chevron icons are updated in this PR: https://github.com/bluedotimpact/bluedot/pull/1368
